### PR TITLE
3602 common case files

### PR DESCRIPTION
**Pull request by briangsweeney**
Early success selecting the data and pushing it to the UI.  Code quality issues and other refinements lacking.

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java]: Please copy paste the Autopsy header with a single 2018 copyright date.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        this.setEnabled(Case.isCaseOpen());
+        Case.addEventTypeSubscriber(EnumSet.of(Case.Events.CURRENT_CASE), (PropertyChangeEvent evt) -> {
+            if (evt.getPropertyName().equals(Case.Events.CURRENT_CASE.toString())) {
+                setEnabled(evt.getNewValue() != null);
+            }
+        });
+    }
+    
+    public static CommonFilesAction getDefault(){
+        if(instance == null){
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java]: Please do a format of all files before committing (ALT+SHIFT+F).

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesChildren.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    CommonFilesChildren(boolean lazy, List<AbstractFile> fileList){
+        super(lazy);
+        this.setKeys(fileList);
+    }
+    
+    @Override
+    protected Node[] createNodes(AbstractFile t) {
+        Node[] node = new Node[1];
+        if(t.isDir()){
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesChildren.java]: This check is not necessary, we do not compute MD5 hashes for directories.

File: branding/core/core.jar/org/netbeans/core/startup/Bundle.properties [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #Updated by build script
-#Wed, 28 Feb 2018 09:59:56 -0500
+#Fri, 09 Mar 2018 13:03:41 -0700
```
Comment (rcordovano) [inline branding/core/core.jar/org/netbeans/core/startup/Bundle.properties]: Next time, please remember: git checkout -- branding

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesDialog.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.JFrame;
+import org.openide.util.NbBundle;
+import org.openide.windows.WindowManager;
+
+/**
+ *
+ * @author bsweeney
+ */
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesDialog.java]: We do not do these types of class headers, please remove authorship and provide description.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesDialog.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+public class CommonFilesDialog extends javax.swing.JDialog {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesDialog.java]: Please make all classes not intended to be base classes final.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    void addListenerToAll(ActionListener l) {
+        this.searchButton.addActionListener(l);
+    }
+    
+    private void search(){
+        
+        String title = NbBundle.getMessage(this.getClass(), "CommonFilesPanel.search.results.title");
+        String pathText = NbBundle.getMessage(this.getClass(), "CommonFilesPanel.search.results.pathText");
+
+        //TODO somehow thats not deprecated???
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java]: This method is definitely deprecated. Perhaps you need to merge in upstream/develop. The replacement method is Case.getOpenCase, which throws an exception.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        Case currentCase = Case.getCurrentCase();
+               
+        List<AbstractFile> contentList = null;
+        try {
+            SleuthkitCase tskDb = currentCase.getSleuthkitCase();
+            //tskDb.executeQuery("SELECT *, COUNT(*) FROM  tsk_files GROUP BY  md5 HAVING  COUNT(*) > 1;");
+            contentList = tskDb.findAllFilesWhere("1 == 1 GROUP BY  md5 HAVING  COUNT(*) > 1;");
+        } catch (TskCoreException ex) {
+            Logger logger = Logger.getLogger(this.getClass().getName());
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java]: Please define a Logger and store the reference as a final class (static) field.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/Bundle.properties [hunk 9]
```diff
@@ -0,0 +1,7 @@
+CommonFilesNode.getName.text=Common Files
+CommonFilesDialog.frame.title=Find Common Files
+CommonFilesDialog.frame.msg=Find Common Files
+CommonFilesPanel.searchButton.text=Search
+CommonFilesPanel.search.results.title=Common Files
+CommonFilesPanel.search.results.pathText=Common Files Search Results\:
+CommonFilesAction.getName.text=Common Files Search
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/Bundle.properties]: Please use @Message annotations for all messages not generated by the IDE UI design tool (Matisse).

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesNode.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2011 Basis Technology Corp.
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesNode.java]: Please change copyright to 2018.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesChildren.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2011 Basis Technology Corp.
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesChildren.java]: Please change copyright to 2018.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesDialog.java [hunk 12]
```diff
@@ -0,0 +1,79 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesDialog.java]: Please copy paste the Autopsy header with a single 2018 copyright date.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+import org.openide.util.HelpCtx;
+import org.openide.util.NbBundle;
+import org.openide.util.actions.CallableSystemAction;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.directorytree.FileSearchProvider;
+
+/**
+ *
+ * @author bsweeney
+ */
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java]: We do not do these types of class headers, please remove authorship and provide description.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+final class CommonFilesAction extends CallableSystemAction implements FileSearchProvider {
+
+    private static CommonFilesAction instance = null;
+    
+    CommonFilesAction(){
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java]: Throughout, Javadocs (type /** to start) for methods please.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+        super();
+        this.setEnabled(Case.isCaseOpen());
+        Case.addEventTypeSubscriber(EnumSet.of(Case.Events.CURRENT_CASE), (PropertyChangeEvent evt) -> {
+            if (evt.getPropertyName().equals(Case.Events.CURRENT_CASE.toString())) {
+                setEnabled(evt.getNewValue() != null);
+            }
+        });
+    }
+    
+    public static CommonFilesAction getDefault(){
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesAction.java]: I think this should be synchronized, but is it required?

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesNode.java [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.List;
+import org.openide.nodes.AbstractNode;
+import org.openide.nodes.Children;
+import org.openide.util.NbBundle;
+import org.sleuthkit.datamodel.AbstractFile;
+
+/**
+ *
+ * @author bsweeney
+ */
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesNode.java]: We do not do these types of class headers, please remove authorship and provide description.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/Bundle.properties (hunk 9 above)
Comment (briangsweeney) [reply Core/src/org/sleuthkit/autopsy/commonfilesearch/Bundle.properties]: Could you provide an example, I don't follow?

Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/commonfilesearch/Bundle.properties]: @briangsweeney, is the the SwingWorker code sample I emailed out as SwingWorker_Snippet_2.text sufficient as an example?

Comment (briangsweeney) [reply Core/src/org/sleuthkit/autopsy/commonfilesearch/Bundle.properties]: got it now. thx!

File: Core/src/org/sleuthkit/autopsy/datamodel/CommonFileNode.java [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.datamodel;
+
+import org.sleuthkit.datamodel.AbstractFile;
+
+/**
+ *
+ * @author bsweeney
+ */
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/datamodel/CommonFileNode.java]: Please replace the author attribution (we don't use them) with a description of the class.

File: Core/src/org/sleuthkit/autopsy/datamodel/CommonFileNode.java [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+public class CommonFileNode extends AbstractFsContentNode<AbstractFile> {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/datamodel/CommonFileNode.java]: This class does not need to extend AbstractFsContentNode<AbstractFile>. That base class is only needed for nodes in the tree view. The nice thing about this is that you will not need to implement the ContentNode and DisplayableItemNode interface methods, which are currently throwing unsupportedOperationException. Instead, you need to provide an override of AbstractFile.createSheet.

**Comment by briangsweeney** [comment]
these issues have been addressed on a separate branch (3602-common-case-files_r2) and will be closed as part of a separate pull request when that code is stable.

**Pull request closed without merging**